### PR TITLE
TST: added test to ensure subclasses of ExcelWriter don't add any public attributes

### LINKED ISSUE #49602: TST: Subclasses of ExcelWriter should not be allowed to add new public methods
Ref: #48577

Subclasses of ExcelWriter (e.g. XlsxWriter) should not be allowed to add any public methods so that the methods accessible by users are independent of the engine being used. This appears to currently be the case, but we should add tests for each subclass that would fail if any additional public methods are added.

### PR DESCRIPTION
**Pull request by catmar22**
- [ ] closes #49602
- not fully tested with style and pre-commit checks yet

### CODE REVIEW

File: pandas/tests/io/excel/test_writers.py [hunk 1]
```diff
@@ -30,6 +30,9 @@
     register_writer,
 )
 
+# added for last test
```
Comment (mroeschke) [inline pandas/tests/io/excel/test_writers.py:33]: This comment isn't needed

File: pandas/tests/io/excel/test_writers.py [hunk 2]
```diff
@@ -1353,3 +1356,12 @@ def test_excelwriter_fspath(self):
...
                 assert os.fspath(writer) == str(path)
+
+# testing that subclasses of ExcelWriter don't have public attributes (issue 49602)
+
+
+@pytest.mark.parametrize("klass", _writers.values())
+def test_subclass_attr(klass):
+    attrs_base = {name for name in dir(ExcelWriter) if not name.startswith("_")}
+    attrs_klass = {name for name in dir(klass) if not name.startswith("_")}
+    assert attrs_base.symmetric_difference(attrs_klass) == set()
```
Comment (mroeschke) [inline pandas/tests/io/excel/test_writers.py:1367]:
```suggestion
    assert not attrs_base.symmetric_difference(attrs_klass)
```

**Comment by catmar22** [comment]
I am failing the pre-commit check and from what I saw in the results of the tests this time, I think it has to do with the import statement? Do you have any input on what the issue might be, ie did I put it in a bad spot based on certain code style/maintenance criteria? Or where I could look for further documentation on this?

**Comment by mroeschke** [comment]
https://pandas.pydata.org/docs/development/contributing_codebase.html#pre-commit

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @catmar22

**Comment by rhshadrach** [comment]
Thanks @catmar22!

**Pull request merged**